### DISCUSSION TITLE
CPU Usage

### DISCUSSION THREAD

**Discussion by IllNate** [thorrak/fermentrack#651] [General]
I was running fermentrack on a rpi4 and loved it. But I was also using the pi to run a weather station and pihole. I was having trouble with it freezing up and ended up uninstalling fermentrack, which resolved the issue, but I've missed it dearly and have recently installed it in a docker container on my unraid server. Works great, but I'm seeing 25% CPU usage. For comparisson, I have another container running shinobi which records my CTV cameras (2@1440p) feeds 24hrs a day, and this only takes up 10% CPU. Thinking back to the rpi4, I seem to recall that it also hogged an entire cpu thread. Has anyone else noticed this?

**Comment by lbussy** [comment]
Doesn't seem right:

```
top - 16:09:47 up 5 days, 44 min,  3 users,  load average: 0.11, 0.03, 0.01
Tasks: 145 total,   2 running, 143 sleeping,   0 stopped,   0 zombie
%Cpu(s):  0.7 us,  0.3 sy,  0.0 ni, 99.0 id,  0.0 wa,  0.0 hi,  0.0 si,  0.0 st
MiB Mem :   1002.8 total,    111.7 free,    393.3 used,    497.7 buff/cache
MiB Swap:   1075.0 total,   1052.0 free,     23.0 used.    443.8 avail Mem

  PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
 1678 statd     20   0   47124  19844  10180 R   1.3   1.9 101:09.81 python
  450 root      20   0  978096  33504   9024 S   0.3   3.3   8:27.30 dockerd
 1676 statd     20   0  120508  52828  13492 S   0.3   5.1   9:32.67 python
12821 root      20   0       0      0      0 I   0.3   0.0   0:05.65 kworker/0+
28969 pi        20   0   12436   4576   3764 S   0.3   0.4   0:00.01 sshd
    1 root      20   0   34532   6940   5228 S   0.0   0.7   0:03.24 systemd
    2 root      20   0       0      0      0 S   0.0   0.0   0:00.02 kthreadd
```

**Comment by IllNate** [comment]
I see the problem on the docker container, it's creating an infinite log of this: 
` Jul 19 2021 22:29:41   Refreshing dbConfig`
` Jul 19 2021 22:29:41   Another instance of BrewPi is already running, which will conflict with this instance. This instance will exit`

**Reply by IllNate** [reply]
After restarting the docker container, it stopped doing this. I had a power outage yesterday, so perhaps there's a lockout or something that didn't get released. CPU usage still spiked to 25%, but seems like only once per minute now, which is acceptable.

**Comment by wmbtwarrior** [comment]
Hello, i know this has been 4 years later and you might not even use fermentrack anymore, but what container did you have installed for unraid? i cant for the life of me find a container in community apps, and when i try to install from docker hub it fails since thorrak doesnt have a repository in it.

**Comment by IllNate** [comment]
Yeah sorry. I actually converted my iSpindels into ESPHOME devices and use home assistant to track my gravity. But I haven't even done that since I've been fermented in stainless fermenters. I modified one of my iSpindels to use a whip antenna instead of the chip antenna, which did increase the transmit power, but still can't penetrate the fermenter. 

If I recall how I got it running on Unraid, I think I just used docker compose directly.

**Comment by lalo-uy** [comment]
I used an old wifi router as AP siting over the tank to get the signal in.

**Comment by uSlackr** [comment]
ESPHome provides a bluetooth proxy to relay bt data into home assistant.  I'm using the Tilt Pro to get data out of the tank with the proxy located above the tank.  It's working in a pro setting as well as my basement.  Well, at home I'm not using a Pro device - just standard Tilt.

I suspect a similar device could be developed to relay ispindel data over wifi.

\\Greg
________________________________

**Comment by wmbtwarrior** [comment]
No worries I figured it was a long shot. I have had a awful time with
docker compose for some reason. I am also pretty darn green with them so
every attempt has failed. Doesn't help motivation lol. Thanks sir!

**Reply by IllNate** [reply]
I guess, I uploaded it to github before I uninstalled it. Not sure if it'll work currently or not:
https://github.com/IllNate/fermentrack-docker-unraid

**Reply by wmbtwarrior** [reply]
Oh that is awesome! My problem is i cant seem to figure out how to create a container if the repository is not apart of community apps.  I have searched up how to add a repository to CA, and they all say to click on either Docker tab, then Docker Repositories sub tab to add one, or to go to APPS and setting to add a repository however i can not find either of those in unraid 7.0.1.

**Reply by IllNate** [reply]
You might want to check out the "Compose Manager" plugin. That's what I used, but again, haven't been using this lately.

**Reply by wmbtwarrior** [reply]
i will give that a try, thanks sir!

**Comment by uSlackr** [comment]
Mind sharing the esphome config you are using?  I've been doing a lot of esphome and have a iSpindel sitting here for testing.  Perhaps over PM. Not sure it is relevant here.

\\Greg
________________________________

**Reply by IllNate** [reply]
https://github.com/IllNate/ispindel-esphome
You'll need to replace the secretes or create your own secrets file.

**Comment by thorrak** [comment]
Feel free to post it here - it’s off topic, but that’s fine

Sent from my iPhone

On Sep 15, 2025, at 12:10 PM, uSlackr ***@***.***> wrote:

﻿

Mind sharing the esphome config you are using? I've been doing a lot of esphome and have a iSpindel sitting here for testing. Perhaps over PM. Not sure it is relevant here.

\\Greg
________________________________
